### PR TITLE
Change from dynamic generic to static generic，update readme, and avoid memory copying during send and recv.

### PR DESCRIPTION
**Pull request by GTwhy**

### CODE REVIEW

File: src/agent.rs [hunk 1]
```diff
@@ -189,27 +189,16 @@ impl Agent {
...
-            .allocator
-            // alignment 1 is always correct
-            .alloc(unsafe { &Layout::from_size_align_unchecked(data.1, 1) })?;
-        res_lmr.as_mut_slice().copy_from_slice(
-            data.0
-                .get(*SEND_DATA_OFFSET..(SEND_DATA_OFFSET.overflow_add(data.1)))?
-                .as_slice(),
-        );
-        Ok(res_lmr)
+        lmr.take(0..len)
```
Comment (rogercloud) [inline src/agent.rs:201]: Will the length of original lmr be lost?

Comment (GTwhy) [reply src/agent.rs:201]: yes, and we don't need users know the meaningless content in range [len..] of the original lmr. So I think lost the original length is ok.

Comment (GTwhy) [reply src/agent.rs:201]: Local mr is just a wrapper of raw mr for now, and the length of raw mr won't be lost, so losing lmr's length does not cause a memory leak.

**Comment by GTwhy** [comment]
fmt rust code in readme.md and rebase just now

File: src/agent.rs (hunk 1 above)
Comment (rogercloud) [reply src/agent.rs:201]: Does it effect the LocalMr release operation? Especially the allocator function.

Comment (GTwhy) [reply src/agent.rs:201]: No, when LocalMr release, the related memory space will be released according to the `addr` and `len` of RawMr which referenced by the releasing LocalMr.

**Comment by GTwhy** [comment]
pub take() -> pub(crate) take()

**Review by rogercloud — APPROVED** [review]

**Pull request merged**